core/Enclave/enclave_hsm.cpp

```diff
@@ -43,767 +42,611 @@
-        0x85, 0xd0, 0x3a, 0x62, 0x37, 0x30, 0xae, 0xad,
-        0x3e, 0x3d, 0xaa, 0xee, 0x9c, 0x60, 0x73, 0x1d,
-        0xb0, 0x5b, 0xe8, 0x62, 0x1c, 0x4b, 0xeb, 0x38
-    },
-    {
-        0xd4, 0x81, 0x40, 0xd9, 0x50, 0xe2, 0x57, 0x7b,
-        0x26, 0xee, 0xb7, 0x41, 0xe7, 0xc6, 0x14, 0xe2,
-        0x24, 0xb7, 0xbd, 0xc9, 0x03, 0xf2, 0x9a, 0x28,
-        0xa8, 0x3c, 0xc8, 0x10, 0x11, 0x14, 0x5e, 0x06
-    }
+    {0x72, 0x12, 0x8a, 0x7a, 0x17, 0x52, 0x6e, 0xbf,
+     0x85, 0xd0, 0x3a, 0x62, 0x37, 0x30, 0xae, 0xad,
+     0x3e, 0x3d, 0xaa, 0xee, 0x9c, 0x60, 0x73, 0x1d,
+     0xb0, 0x5b, 0xe8, 0x62, 0x1c, 0x4b, 0xeb, 0x38},
+    {0xd4, 0x81, 0x40, 0xd9, 0x50, 0xe2, 0x57, 0x7b,
+     0x26, 0xee, 0xb7, 0x41, 0xe7, 0xc6, 0x14, 0xe2,
+     0x24, 0xb7, 0xbd, 0xc9, 0x03, 0xf2, 0x9a, 0x28,
+     0xa8, 0x3c, 0xc8, 0x10, 0x11, 0x14, 0x5e, 0x06}
 
 };
 
-void printf(const char *fmt, ...)
-{
-    char buf[BUFSIZ] = {'\0'};
-    va_list ap;
-    va_start(ap, fmt);
-    vsnprintf(buf, BUFSIZ, fmt, ap);
-    va_end(ap);
-    ocall_print_string(buf);
-}
-
-typedef struct _aes_gcm_data_ex_t
-{
-    uint32_t  ciphertext_size;
-    uint32_t  aad_size;
-    uint8_t   reserve1[8];
-    uint8_t   iv[SGX_AESGCM_IV_SIZE];
-    uint8_t   reserve2[4];
-    uint8_t   mac[SGX_AESGCM_MAC_SIZE];
-    uint8_t   payload[];   /* ciphertext + aad */
-} sgx_aes_gcm_data_ex_t;
-
-static uint32_t sgx_calc_gcm_data_size(const uint32_t aad_size, const uint32_t plaintext_size)
-{
-    if (aad_size > UINT32_MAX - sizeof(sgx_aes_gcm_data_ex_t))
-        return UINT32_MAX;
-
-    if (plaintext_size > UINT32_MAX - sizeof(sgx_aes_gcm_data_ex_t))
-        return UINT32_MAX;
-
-    if (aad_size > UINT32_MAX - plaintext_size)
-        return UINT32_MAX;
-
-    if (sizeof(sgx_aes_gcm_data_ex_t) > UINT32_MAX - plaintext_size - aad_size)
-        return UINT32_MAX;
-
-    return (aad_size + plaintext_size + sizeof(sgx_aes_gcm_data_ex_t));
-}
-
-static uint32_t sgx_get_gcm_ciphertext_size(const sgx_aes_gcm_data_ex_t *gcm_data)
-{
-    if (NULL == gcm_data)
-        return UINT32_MAX;
-
-    return gcm_data->ciphertext_size;
-}
-
-
-static sgx_status_t sgx_gcm_encrypt(const sgx_aes_gcm_128bit_key_t *key,
-                                    const uint32_t plaintext_size, const uint8_t *plaintext,
-                                    const uint32_t aad_size, const uint8_t *aad,
-                                    const uint32_t gcm_data_size, sgx_aes_gcm_data_ex_t *gcm_data)
+sgx_status_t enclave_create_key(ehsm_keyblob_t *cmk, size_t cmk_len)
 {
-    if (NULL == gcm_data || gcm_data_size < sgx_calc_gcm_data_size(aad_size, plaintext_size))
-        return SGX_ERROR_INVALID_PARAMETER;
-
-    if (plaintext_size == 0 || NULL == plaintext)
-        return SGX_ERROR_INVALID_PARAMETER;
+    sgx_status_t ret = SGX_ERROR_UNEXPECTED;
 
-    if (plaintext_size > UINT32_MAX - aad_size)
+    if (cmk == NULL || cmk->metadata.origin != EH_INTERNAL_KEY)
+    {
```

Comment:
it's the entry of th ecall, you need to add more check here

Reply:
check the cmk's length

Reply:
copy